### ISSUE TITLE
Products not loading in the app

### ISSUE THREAD

**Issue by joashrajin** [woocommerce/woocommerce-ios#9369]
**Describe the bug**

We have a case reported in 6074239-zd-woothemes that we could not reproduce where the app was unable to load any products. In the app logs, we notice:

```
2023/03/27 00:52:01:851  ⛔️ Error synchronizing products: typeMismatch(Swift.Array<Any>, Swift.DecodingError.Context(codingPath: [], debugDescription: "Expected to decode Array<Any> but found a dictionary instead.", underlyingError: nil))

```

To Reproduce: 

- Open the Products Page
- Scroll down to the bottom so more products try to load

**Isolating the problem (mark completed items with an [x]):**
- [x] I have deactivated other plugins and confirmed this bug occurs when only WooCommerce plugin is active.

**Mobile Environment**
Please include:
- Device: iPhone 12
- iOS version: 16.3.1
- WooCommerce iOS version: 12.6.0.2

**WordPress Environment**
<details>
```
### WordPress Environment ###

WordPress address (URL): ----redacted----
Site address (URL): ----redacted----
WC Version: 7.5.1
REST API Version: ✔ 7.5.1
WC Blocks Version: ✔ 9.9.0
Action Scheduler Version: ✔ 3.5.4
Log Directory Writable: ✔
WP Version: 6.2
WP Multisite: –
WP Memory Limit: 512 MB
WP Debug Mode: –
WP Cron: ✔
Language: en_US
External object cache: –

### Server Environment ###

Server Info: Apache/2.4.52 (Unix) OpenSSL/1.1.1n
PHP Version: 7.4.27
PHP Post Max Size: 80 MB
PHP Time Limit: 30
PHP Max Input Vars: 1000
cURL Version: 7.64.0
OpenSSL/1.1.1n

SUHOSIN Installed: –
MySQL Version: 5.5.5-10.3.32-MariaDB
Max Upload Size: 80 MB
Default Timezone is UTC: ✔
fsockopen/cURL: ✔
SoapClient: ✔
DOMDocument: ✔
GZip: ✔
Multibyte String: ✔
Remote Post: ✔
Remote Get: ✔

### Database ###

WC Database Version: 7.5.1
WC Database Prefix: wp_
Total Database Size: 69.44MB
Database Data Size: 61.62MB
Database Index Size: 7.82MB
wp_woocommerce_sessions: Data: 2.02MB + Index: 0.05MB + Engine InnoDB
wp_woocommerce_api_keys: Data: 0.02MB + Index: 0.03MB + Engine InnoDB
wp_woocommerce_attribute_taxonomies: Data: 0.02MB + Index: 0.02MB + Engine InnoDB
wp_woocommerce_downloadable_product_permissions: Data: 0.02MB + Index: 0.06MB + Engine InnoDB
wp_woocommerce_order_items: Data: 0.08MB + Index: 0.02MB + Engine InnoDB
wp_woocommerce_order_itemmeta: Data: 0.47MB + Index: 0.39MB + Engine InnoDB
wp_woocommerce_tax_rates: Data: 0.02MB + Index: 0.06MB + Engine InnoDB
wp_woocommerce_tax_rate_locations: Data: 0.02MB + Index: 0.03MB + Engine InnoDB
wp_woocommerce_shipping_zones: Data: 0.02MB + Index: 0.00MB + Engine InnoDB
wp_woocommerce_shipping_zone_locations: Data: 0.02MB + Index: 0.03MB + Engine InnoDB
wp_woocommerce_shipping_zone_methods: Data: 0.02MB + Index: 0.00MB + Engine InnoDB
wp_woocommerce_payment_tokens: Data: 0.02MB + Index: 0.02MB + Engine InnoDB
wp_woocommerce_payment_tokenmeta: Data: 0.02MB + Index: 0.03MB + Engine InnoDB
wp_woocommerce_log: Data: 0.02MB + Index: 0.02MB + Engine InnoDB
wp_actionscheduler_actions: Data: 2.28MB + Index: 1.08MB + Engine InnoDB
wp_actionscheduler_claims: Data: 0.02MB + Index: 0.02MB + Engine InnoDB
wp_actionscheduler_groups: Data: 0.02MB + Index: 0.02MB + Engine InnoDB
wp_actionscheduler_logs: Data: 3.41MB + Index: 0.67MB + Engine InnoDB
wp_commentmeta: Data: 0.02MB + Index: 0.03MB + Engine InnoDB
wp_comments: Data: 0.17MB + Index: 0.09MB + Engine InnoDB
wp_etcpf_category_mappings: Data: 0.02MB + Index: 0.00MB + Engine InnoDB
wp_etcpf_custom_feed_products: Data: 0.02MB + Index: 0.00MB + Engine InnoDB
wp_etcpf_etsy_configuration: Data: 0.02MB + Index: 0.00MB + Engine InnoDB
wp_etcpf_etsy_product_count: Data: 0.02MB + Index: 0.00MB + Engine InnoDB
wp_etcpf_etsy_sync: Data: 0.02MB + Index: 0.00MB + Engine InnoDB
wp_etcpf_feedproducts: Data: 0.02MB + Index: 0.00MB + Engine InnoDB
wp_etcpf_feeds: Data: 0.02MB + Index: 0.00MB + Engine InnoDB
wp_etcpf_image_links: Data: 0.02MB + Index: 0.00MB + Engine InnoDB
wp_etcpf_listings: Data: 0.02MB + Index: 0.00MB + Engine InnoDB
wp_etcpf_listing_variations: Data: 0.02MB + Index: 0.00MB + Engine InnoDB
wp_etcpf_orders: Data: 0.02MB + Index: 0.02MB + Engine InnoDB
wp_etcpf_profiles: Data: 0.02MB + Index: 0.00MB + Engine InnoDB
wp_etcpf_resolved_product_data: Data: 0.02MB + Index: 0.00MB + Engine InnoDB
wp_etcpf_settings: Data: 0.02MB + Index: 0.00MB + Engine InnoDB
wp_etcpf_shipping_template: Data: 0.02MB + Index: 0.00MB + Engine InnoDB
wp_etcpf_variationupload_preparation: Data: 0.02MB + Index: 0.00MB + Engine InnoDB
wp_fsmpt_email_logs: Data: 7.02MB + Index: 0.00MB + Engine InnoDB
wp_gla_budget_recommendations: Data: 0.22MB + Index: 0.14MB + Engine InnoDB
wp_gla_merchant_issues: Data: 0.02MB + Index: 0.00MB + Engine InnoDB
wp_gla_shipping_rates: Data: 0.02MB + Index: 0.05MB + Engine InnoDB
wp_gla_shipping_times: Data: 0.02MB + Index: 0.02MB + Engine InnoDB
wp_koko_analytics_post_stats: Data: 0.11MB + Index: 0.00MB + Engine InnoDB
wp_koko_analytics_referrer_stats: Data: 0.02MB + Index: 0.00MB + Engine InnoDB
wp_koko_analytics_referrer_urls: Data: 0.02MB + Index: 0.02MB + Engine InnoDB
wp_koko_analytics_site_stats: Data: 0.02MB + Index: 0.00MB + Engine InnoDB
wp_links: Data: 0.02MB + Index: 0.02MB + Engine InnoDB
wp_mailpoet_custom_fields: Data: 0.02MB + Index: 0.02MB + Engine InnoDB
wp_mailpoet_dynamic_segment_filters: Data: 0.02MB + Index: 0.02MB + Engine InnoDB
wp_mailpoet_feature_flags: Data: 0.02MB + Index: 0.02MB + Engine InnoDB
wp_mailpoet_forms: Data: 0.02MB + Index: 0.00MB + Engine InnoDB
wp_mailpoet_log: Data: 0.02MB + Index: 0.00MB + Engine InnoDB
wp_mailpoet_mapping_to_external_entities: Data: 0.02MB + Index: 0.02MB + Engine InnoDB
wp_mailpoet_newsletters: Data: 0.02MB + Index: 0.03MB + Engine InnoDB
wp_mailpoet_newsletter_links: Data: 0.02MB + Index: 0.05MB + Engine InnoDB
wp_mailpoet_newsletter_option: Data: 0.02MB + Index: 0.02MB + Engine InnoDB
wp_mailpoet_newsletter_option_fields: Data: 0.02MB + Index: 0.02MB + Engine InnoDB
wp_mailpoet_newsletter_posts: Data: 0.02MB + Index: 0.02MB + Engine InnoDB
wp_mailpoet_newsletter_segment: Data: 0.02MB + Index: 0.02MB + Engine InnoDB
wp_mailpoet_newsletter_templates: Data: 2.52MB + Index: 0.00MB + Engine InnoDB
wp_mailpoet_scheduled_tasks: Data: 0.02MB + Index: 0.03MB + Engine InnoDB
wp_mailpoet_scheduled_task_subscribers: Data: 0.02MB + Index: 0.02MB + Engine InnoDB
wp_mailpoet_segments: Data: 0.02MB + Index: 0.03MB + Engine InnoDB
wp_mailpoet_sending_queues: Data: 0.02MB + Index: 0.03MB + Engine InnoDB
wp_mailpoet_settings: Data: 0.02MB + Index: 0.02MB + Engine InnoDB
wp_mailpoet_statistics_bounces: Data: 0.02MB + Index: 0.00MB + Engine InnoDB
wp_mailpoet_statistics_clicks: Data: 0.02MB + Index: 0.05MB + Engine InnoDB
wp_mailpoet_statistics_forms: Data: 0.02MB + Index: 0.02MB + Engine InnoDB
wp_mailpoet_statistics_newsletters: Data: 0.02MB + Index: 0.03MB + Engine InnoDB
wp_mailpoet_statistics_opens: Data: 0.02MB + Index: 0.08MB + Engine InnoDB
wp_mailpoet_statistics_unsubscribes: Data: 0.02MB + Index: 0.05MB + Engine InnoDB
wp_mailpoet_statistics_woocommerce_purchases: Data: 0.02MB + Index: 0.06MB + Engine InnoDB
wp_mailpoet_stats_notifications: Data: 0.02MB + Index: 0.03MB + Engine InnoDB
wp_mailpoet_subscribers: Data: 0.02MB + Index: 0.13MB + Engine InnoDB
wp_mailpoet_subscriber_custom_field: Data: 0.02MB + Index: 0.02MB + Engine InnoDB
wp_mailpoet_subscriber_ips: Data: 0.02MB + Index: 0.02MB + Engine InnoDB
wp_mailpoet_subscriber_segment: Data: 0.02MB + Index: 0.03MB + Engine InnoDB
wp_mailpoet_user_agents: Data: 0.02MB + Index: 0.02MB + Engine InnoDB
wp_mailpoet_user_flags: Data: 0.02MB + Index: 0.02MB + Engine InnoDB
wp_options: Data: 4.02MB + Index: 0.16MB + Engine InnoDB
wp_postmeta: Data: 4.41MB + Index: 1.86MB + Engine InnoDB
wp_posts: Data: 13.06MB + Index: 0.31MB + Engine InnoDB
wp_redirects: Data: 0.02MB + Index: 0.00MB + Engine InnoDB
wp_snippets: Data: 0.02MB + Index: 0.03MB + Engine InnoDB
wp_stock_log: Data: 0.05MB + Index: 0.00MB + Engine InnoDB
wp_termmeta: Data: 0.02MB + Index: 0.03MB + Engine InnoDB
wp_terms: Data: 0.02MB + Index: 0.03MB + Engine InnoDB
wp_term_relationships: Data: 0.03MB + Index: 0.02MB + Engine InnoDB
wp_term_taxonomy: Data: 0.02MB + Index: 0.03MB + Engine InnoDB
wp_usermeta: Data: 0.27MB + Index: 0.14MB + Engine InnoDB
wp_users: Data: 0.02MB + Index: 0.05MB + Engine InnoDB
wp_wc_admin_notes: Data: 0.05MB + Index: 0.00MB + Engine InnoDB
wp_wc_admin_note_actions: Data: 0.02MB + Index: 0.02MB + Engine InnoDB
wp_wc_category_lookup: Data: 0.02MB + Index: 0.00MB + Engine InnoDB
wp_wc_customer_lookup: Data: 0.02MB + Index: 0.03MB + Engine InnoDB
wp_wc_download_log: Data: 0.02MB + Index: 0.03MB + Engine InnoDB
wp_wc_order_bundle_lookup: Data: 0.02MB + Index: 0.09MB + Engine InnoDB
wp_wc_order_coupon_lookup: Data: 0.02MB + Index: 0.03MB + Engine InnoDB
wp_wc_order_product_lookup: Data: 0.06MB + Index: 0.06MB + Engine InnoDB
wp_wc_order_stats: Data: 0.02MB + Index: 0.05MB + Engine InnoDB
wp_wc_order_tax_lookup: Data: 0.02MB + Index: 0.03MB + Engine InnoDB
wp_wc_product_attributes_lookup: Data: 0.02MB + Index: 0.02MB + Engine InnoDB
wp_wc_product_download_directories: Data: 0.02MB + Index: 0.02MB + Engine InnoDB
wp_wc_product_meta_lookup: Data: 0.02MB + Index: 0.09MB + Engine InnoDB
wp_wc_rate_limits: Data: 0.02MB + Index: 0.02MB + Engine InnoDB
wp_wc_reserved_stock: Data: 0.02MB + Index: 0.00MB + Engine InnoDB
wp_wc_tax_rate_classes: Data: 0.02MB + Index: 0.02MB + Engine InnoDB
wp_wc_webhooks: Data: 0.02MB + Index: 0.02MB + Engine InnoDB
wp_wfblockediplog: Data: 0.02MB + Index: 0.00MB + Engine InnoDB
wp_wfblocks7: Data: 0.02MB + Index: 0.05MB + Engine InnoDB
wp_wfconfig: Data: 1.38MB + Index: 0.00MB + Engine InnoDB
wp_wfcrawlers: Data: 0.02MB + Index: 0.00MB + Engine InnoDB
wp_wffilechanges: Data: 0.02MB + Index: 0.00MB + Engine InnoDB
wp_wffilemods: Data: 10.52MB + Index: 0.00MB + Engine InnoDB
wp_wfhits: Data: 1.30MB + Index: 0.05MB + Engine InnoDB
wp_wfhoover: Data: 0.02MB + Index: 0.02MB + Engine InnoDB
wp_wfissues: Data: 0.02MB + Index: 0.06MB + Engine InnoDB
wp_wfknownfilelist: Data: 5.52MB + Index: 0.00MB + Engine InnoDB
wp_wflivetraffichuman: Data: 0.02MB + Index: 0.02MB + Engine InnoDB
wp_wflocs: Data: 0.02MB + Index: 0.00MB + Engine InnoDB
wp_wflogins: Data: 0.02MB + Index: 0.03MB + Engine InnoDB
wp_wfls_2fa_secrets: Data: 0.02MB + Index: 0.02MB + Engine InnoDB
wp_wfls_role_counts: Data: 0.00MB + Index: 0.00MB + Engine MEMORY
wp_wfls_settings: Data: 0.02MB + Index: 0.00MB + Engine InnoDB
wp_wfnotifications: Data: 0.02MB + Index: 0.00MB + Engine InnoDB
wp_wfpendingissues: Data: 0.02MB + Index: 0.06MB + Engine InnoDB
wp_wfreversecache: Data: 0.02MB + Index: 0.00MB + Engine InnoDB
wp_wfsnipcache: Data: 0.02MB + Index: 0.05MB + Engine InnoDB
wp_wfstatus: Data: 0.11MB + Index: 0.09MB + Engine InnoDB
wp_wftrafficrates: Data: 0.02MB + Index: 0.00MB + Engine InnoDB
wp_woocommerce_bundled_itemmeta: Data: 0.06MB + Index: 0.06MB + Engine InnoDB
wp_woocommerce_bundled_items: Data: 0.02MB + Index: 0.03MB + Engine InnoDB
wp_woocommerce_square_customers: Data: 0.02MB + Index: 0.00MB + Engine InnoDB
wp_wpfm_backup: Data: 0.02MB + Index: 0.00MB + Engine InnoDB
wp_wpmailsmtp_debug_events: Data: 0.02MB + Index: 0.00MB + Engine InnoDB
wp_wpmailsmtp_tasks_meta: Data: 0.02MB + Index: 0.00MB + Engine InnoDB
wp_yoast_indexable: Data: 0.16MB + Index: 0.09MB + Engine InnoDB
wp_yoast_indexable_hierarchy: Data: 0.02MB + Index: 0.05MB + Engine InnoDB
wp_yoast_migrations: Data: 0.02MB + Index: 0.02MB + Engine InnoDB
wp_yoast_primary_term: Data: 0.02MB + Index: 0.03MB + Engine InnoDB
wp_yoast_seo_links: Data: 0.06MB + Index: 0.03MB + Engine InnoDB

### Post Type Counts ###

attachment: 522
custom_css: 1
mailpoet_page: 1
nav_menu_item: 18
nectar_slider: 3
page: 10
post: 54
product: 57
product_variation: 48
revision: 788
shop_coupon: 3
shop_order: 104
shop_order_refund: 9
wp_global_styles: 1
yith_wcan_preset: 1
yith-wcbm-badge: 2

### Security ###

Secure connection (HTTPS): ✔
Hide errors from visitors: ✔

### Active Plugins (32) ###

Koko Analytics: by ibericode – 1.0.37
FiboSearch - AJAX Search for WooCommerce: by FiboSearch Team – 1.22.3
Classic Widgets: by WordPress Contributors – 0.3
Code Snippets: by Code Snippets Pro – 3.3.0
Colorlib Login Customizer: by Colorlib – 1.3.1
Yoast Duplicate Post: by Enrico Battocchi & Team Yoast – 4.5
301 Redirects: by WebFactory Ltd – 2.73
FluentSMTP: by FluentSMTP & WPManageNinja Team – 2.2.4
Jetpack: by Automattic – 11.9.1
Salient WPBakery Page Builder: by Michael M - WPBakery.com | Modified by ThemeNectar – 6.9.1
Salient Core: by ThemeNectar – 1.9.8
Salient Demo Importer: by ThemeNectar – 1.5
Salient Home Slider: by ThemeNectar – 1.4.1
Salient Nectar Slider: by ThemeNectar – 1.7.3
Salient Portfolio: by ThemeNectar – 1.7.2
Salient Shortcodes: by ThemeNectar – 1.5.1
Salient Social: by ThemeNectar – 1.2.2
Salient Widgets: by ThemeNectar – 1.3
Variation Swatches for WooCommerce: by CartFlows – 1.0.7
WooCommerce Blocks: by Automattic – 9.9.0
WooCommerce PayPal Payments: by WooCommerce – 2.0.3
WooCommerce Product Bundles: by WooCommerce – 6.15.2
WooCommerce Shipping & Tax: by WooCommerce – 2.2.4
WooCommerce USPS Shipping: by WooCommerce – 4.4.70
WooCommerce Square: by WooCommerce – 3.6.1
Stock Manager for WooCommerce: by StoreApps – 2.10.0
WooCommerce: by Automattic – 7.5.1
Wordfence Login Security: by Wordfence – 1.1.2
Wordfence Security: by Wordfence – 7.9.2
Yoast SEO: by Team Yoast – 20.4
YITH WooCommerce Ajax Product Filter: by YITH – 4.20.0
YITH WooCommerce Badge Management: by YITH – 2.15.0

### Inactive Plugins (0) ###

### Settings ###

API Enabled: –
Force SSL: ✔
Currency: USD ($)
Currency Position: left
Thousand Separator: ,
Decimal Separator: .
Number of Decimals: 2
Taxonomies: Product Types: bundle (bundle)
external (external)
grouped (grouped)
simple (simple)
variable (variable)

Taxonomies: Product Visibility: exclude-from-catalog (exclude-from-catalog)
exclude-from-search (exclude-from-search)
featured (featured)
outofstock (outofstock)
rated-1 (rated-1)
rated-2 (rated-2)
rated-3 (rated-3)
rated-4 (rated-4)
rated-5 (rated-5)

Connected to WooCommerce.com: –
Enforce Approved Product Download Directories: –
Order datastore: WC_Order_Data_Store_CPT

### WC Pages ###

Shop base: [#8](8-zd-woothemes) - /market/
Cart: [#9](9-zd-woothemes) - /cart/
Checkout: [#10](10-zd-woothemes) - /checkout/
My account: [#11](11-zd-woothemes) - /my-account/
Terms and conditions: [#5914](5914-zd-woothemes) - /returns-and-shipping/

### Theme ###

Name: Salient
Version: 15.0.8
Author URL: https://themeforest.net/user/themenectar
Child Theme: ❌ – If you are modifying WooCommerce on a parent theme that you did not build personally we recommend using a child theme. See: How to create a child theme
WooCommerce Support: ✔

### Templates ###

Overrides: salient/woocommerce/cart/mini-cart.php
salient/woocommerce/checkout/form-checkout.php
salient/woocommerce/checkout/review-order.php
salient/woocommerce/checkout/terms.php
salient/woocommerce/content-product.php
salient/woocommerce/content-single-product.php
salient/woocommerce/loop/add-to-cart.php
salient/woocommerce/loop/loop-start.php
salient/woocommerce/myaccount/form-login.php
salient/woocommerce/single-product/product-image.php
salient/woocommerce/single-product/rating.php
salient/woocommerce/single-product/tabs/description.php
salient/woocommerce/single-product/tabs/tabs.php
salient/woocommerce/single-product/title.php
salient/woocommerce/single-product-reviews.php

### Product Bundles ###

Database Version: 6.15.2
Loopback Test: ✔
Template Overrides: –

### WooCommerce PayPal Payments ###

Onboarded: ✔
Shop country code: US
WooCommerce currency supported: ✔
Advanced Card Processing available in country: ✔
Pay Later messaging available in country: ✔
Webhook status: ✔
Vault enabled: ✔
Logging enabled: –
Reference Transactions: –
Used PayPal Checkout plugin: –
Tracking enabled: –

### Square ###

Environment: Production
Tokenization Enabled: –
Debug Mode: Off

### Admin ###

Enabled Features: activity-panels
analytics
coupons
customer-effort-score-tracks
import-products-task
experimental-fashion-sample-products
shipping-smart-defaults
shipping-setting-tour
homescreen
marketing
multichannel-marketing
mobile-app-banner
navigation
onboarding
onboarding-tasks
remote-inbox-notifications
remote-free-extensions
payment-gateway-suggestions
shipping-label-banner
subscriptions
store-alerts
transient-notices
woo-mobile-welcome
wc-pay-promotion
wc-pay-welcome-page

Disabled Features: minified-js
new-product-management-experience
product-variation-management
settings

Daily Cron: ✔ Next scheduled: 2023-04-02 02:27:54 -04:00
Options: ✔
Notes: 41
Onboarding: completed

### Action Scheduler ###

Complete: 2,266
Oldest: 2023-03-01 16:28:14 -0500
Newest: 2023-04-01 17:00:04 -0400

Failed: 1,543
Oldest: 2022-04-27 22:53:39 -0400
Newest: 2023-03-26 20:24:00 -0400

Pending: 4
Oldest: 2023-04-01 18:00:04 -0400
Newest: 2023-04-03 17:47:25 -0400

### Status report information ###

Generated at: 2023-04-01 17:23:45 -04:00
```
</details>

**Comment by iamgabrielma** [comment]
```
2023/03/27 00:52:01:851  ⛔️ Error synchronizing products: typeMismatch(Swift.Array<Any>, Swift.DecodingError.Context(codingPath: [], debugDescription: "Expected to decode Array<Any> but found a dictionary instead.", underlyingError: nil))
```

This is generally a plugin conflict: A plugin alters the expected return type of one of the properties from models in core ( possibly the [Products model](https://github.com/woocommerce/woocommerce-ios/blob/trunk/Networking/Networking/Model/Product/Product.swift) in this case ) and returns a dictionary of "something" where the API is expecting an array of "something".

Other examples of a similar issue are:
- https://github.com/woocommerce/woocommerce-ios/issues/4737
- https://github.com/woocommerce/woocommerce-ios/issues/4818
- https://github.com/woocommerce/woocommerce-ios/issues/3511

Can you check with them if the same happens when they do a plugin dance? Taking a quick look to their SSR I'd say possible culprits could be `Code Snippets: by Code Snippets Pro – 3.3.0` , `Variation Swatches for WooCommerce: by CartFlows – 1.0.7`, or `YITH WooCommerce Ajax Product Filter: by YITH – 4.20.0`.  I'm seeing the Product model expects variations to be an array of Int64's, so could be the 2nd plugin on that list, as alters variations directly 🤔

**Comment by joashrajin** [comment]
Thanks, @iamgabrielma :)

I have reached out to the user

**Comment by reginabally** [comment]
We have another case in 6177741-zd-woothemes. The common plugin I saw is `FiboSearch - AJAX Search for WooCommerce`.

From the REST API response for products on this site, it was enclosed in an additional `data` dictionary. So the response looks like this:

```
{
	"data": [
		{
			"id": 66697,
...
  ]
}
```

Edited to add:

It appears that responses returned from the Jetpack will be enclosed in the `data` field. Whereas responses returned from WC REST API don't have it. So, it might be another field that's causing this issue.

Also, apart from the `FiboSearch - AJAX Search for WooCommerce` plugin, I found out the other common plugin for these sites are `Yoast SEO`.

**Comment by reginabally** [comment]
Another report in 6182091-zd-woothemes, mentioned the issue after updating the app to 12.8. I requested the user to perform a plugin conflict test.

List of active plugins on the site:

```
Advanced Local Pickup for WooCommerce: by zorem - 1.5.3
Akismet Anti-Spam: by Automattic - 5.1
Astra Pro: by Brainstorm Force - 4.1.4
Premium Starter Templates: by Brainstorm Force - 3.2.0
Product Feed Manager for WooCommerce: by RexTheme - 7.2.30
Code Snippets: by Code Snippets Pro - 3.3.0
Disable XML-RPC-API: by Neatma - 2.1.4.7
Elementor: by Elementor.com - 3.12.1
Facebook for WooCommerce: by Facebook - 3.0.17
Flying Scripts: by WP Speed Matters - 1.2.3
Forminator Pro: by WPMU DEV - 1.23.2
Site Kit by Google: by Google - 1.96.0
HookMeUp – Unlock and use strategic spots in any WooCommerce Theme: by Get Bowtied - 1.8
WPCode Lite: by WPCode - 2.0.9
Jetpack: by Automattic - 12.0
Mailchimp for WooCommerce: by Mailchimp - 2.8.2
Pinterest for WooCommerce: by WooCommerce - 1.2.17
Route App: by Route - 2.2.4
SVG Support: by Benbodhi - 2.5.5
Ultimate Addons for Elementor: by Brainstorm Force - 1.36.15
Variation Swatches for WooCommerce: by CartFlows - 1.0.7
USPS Simple Shipping for Woocommerce: by dangoodman - 1.8.2
WooCommerce Stripe Gateway: by WooCommerce - 7.2.0
PDF Invoices &amp; Packing Slips for WooCommerce: by WP Overnight - 3.5.0
WooCommerce Shipping &amp; Tax: by WooCommerce - 2.2.4
WooCommerce: by Automattic - 7.5.1
WordPress Importer: by wordpressdotorg - 0.8
Yoast SEO: by Team Yoast - 20.4
Defender Pro: by WPMU DEV - 3.10.0
Hummingbird Pro: by WPMU DEV - 3.4.5
WP Mail SMTP: by WPForms - 3.7.0
Smush Pro: by WPMU DEV - 3.12.6
WPMU DEV Dashboard: by WPMU DEV - 4.11.18
WPS Hide Login: by WPServeur, NicolasKulka, wpformation - 1.9.8
XPS Ship Integration: by XPS Ship - 1.1.72
```

**Comment by rachelmcr** [comment]
@joashrajin @reginabally I'm working on adding some fallbacks to the app so we can try to handle unexpected data caused by plugin conflicts like this. Unfortunately, the errors messages mentioned in this issue are too generic to know exactly what the problem was for these merchants. We released a fix for that in version 13.3 (see #9510 and pe5sF9-1my-p2) so from those versions onward we should have more specific decoding error messages again.

I checked the stores in these user reports and it looks like they are either resolved (no errors that I see with decoding) or Jetpack is no longer connected so we can't check again. If you think it's appropriate, could you follow up with these merchants and see if they are still having any problems using the app, with version 13.3 or later? If so, the app logs should contain the more specific error message.

I'm going to go ahead and close this issue as I don't think there's anything more for us to do with the information we have here. If you do get reports of decoding errors like this from these or other merchants, please do open a new issue (a separate issue for different errors would be great for tracking/resolving them) and we can look into trying to prevent those errors. Thanks!
